components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/configuration/models/APIMConfigurations.java

```diff
@@ -77,6 +77,9 @@
     @Element(description = "rating upper limit")
     private int ratingMaxValue = 5;
 
+    @Element(description = "Set of environments")
+    private EnvironmentConfigurations envs = new EnvironmentConfigurations();
```

Comment:
I think this has to be a list. Otherwise, you can't get more than one environment. Basic idea is that, you have a hardcoded  environment in the code. But when you define environments in the deployment.yaml, you should be able to retrieve those environments.